Build fails: cannot find function `copy_string` in module `crate::clipboard`

Version 0.11.0 fails to build on FreeBSD 12.2 with rust-1.48.0:

```

error[E0425]: cannot find function `copy_string` in module `crate::clipboard`
   --> src/components/commitlist.rs:128:31
    |
128 |             crate::clipboard::copy_string(&e.hash_short)?;
    |                               ^^^^^^^^^^^ not found in `crate::clipboard`

error[E0425]: cannot find function `copy_string` in module `crate::clipboard`
   --> src/components/diff.rs:262:35
    |
262 |                 crate::clipboard::copy_string(
    |                                   ^^^^^^^^^^^ not found in `crate::clipboard`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0425`.
error: could not compile `gitui`
```

Now it fails like this:
```
error[E0432]: unresolved import `which`
  --> src/clipboard.rs:53:9
   |
53 |     use which::which;
   |         ^^^^^ use of undeclared crate or module `which`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `gitui`

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,7 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-<!-- ## Unreleased -->
+## Unreleased
+
+### Fixed
+- compilation broken on freebsd ([#461](https://github.com/extrawurst/gitui/issues/461))
 
 ## [0.11.0] - 2020-12-20
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,7 +41,7 @@ anyhow = "1.0.36"
 unicode-width = "0.1"
 textwrap = "0.13"
 
-[target.'cfg(target_os = "linux")'.dependencies]
+[target.'cfg(all(target_family="unix",not(target_os="macos")))'.dependencies]
 which = "4.0"
 
 # pprof is not available on windows

--- a/src/clipboard.rs
+++ b/src/clipboard.rs
@@ -30,8 +30,7 @@ fn execute_copy_command(command: Command, text: &str) -> Result<()> {
     Ok(())
 }
 
-#[cfg(target_family = "unix")]
-#[cfg(not(target_os = "macos"))]
+#[cfg(all(target_family = "unix", not(target_os = "macos")))]
 fn gen_command(
     path: impl AsRef<OsStr>,
     xclip_syntax: bool,
@@ -46,8 +45,7 @@ fn gen_command(
     c
 }
 
-#[cfg(target_family = "unix")]
-#[cfg(not(target_os = "macos"))]
+#[cfg(all(target_family = "unix", not(target_os = "macos")))]
 pub fn copy_string(string: &str) -> Result<()> {
     use std::path::PathBuf;
     use which::which;

--- a/src/clipboard.rs
+++ b/src/clipboard.rs
@@ -29,7 +29,7 @@ fn execute_copy_command(command: Command, text: &str) -> Result<()> {
     Ok(())
 }
 
-#[cfg(target_os = "linux")]
+#[cfg(target_family = "linux")]
 fn gen_command(
     path: impl AsRef<OsStr>,
     xclip_syntax: bool,
@@ -44,7 +44,7 @@ fn gen_command(
     c
 }
 
-#[cfg(target_os = "linux")]
+#[cfg(target_family = "linux")]
 pub fn copy_string(string: &str) -> Result<()> {
     use std::path::PathBuf;
     use which::which;

--- a/src/clipboard.rs
+++ b/src/clipboard.rs
@@ -1,5 +1,6 @@
 use anyhow::Result;
-#[cfg(target_os = "linux")]
+#[cfg(target_family = "unix")]
+#[cfg(not(target_os = "macos"))]
 use std::ffi::OsStr;
 use std::io::Write;
 use std::process::{Command, Stdio};
@@ -29,7 +30,8 @@ fn execute_copy_command(command: Command, text: &str) -> Result<()> {
     Ok(())
 }
 
-#[cfg(target_family = "linux")]
+#[cfg(target_family = "unix")]
+#[cfg(not(target_os = "macos"))]
 fn gen_command(
     path: impl AsRef<OsStr>,
     xclip_syntax: bool,
@@ -44,7 +46,8 @@ fn gen_command(
     c
 }
 
-#[cfg(target_family = "linux")]
+#[cfg(target_family = "unix")]
+#[cfg(not(target_os = "macos"))]
 pub fn copy_string(string: &str) -> Result<()> {
     use std::path::PathBuf;
     use which::which;

